numpy/core/tests/test_multiarray.py

```diff
@@ -4693,10 +4693,15 @@ def test_unicode_field_titles(self):
             y['a']
 
         def test_unicode_field_names(self):
-            # Unicode field names are not allowed on Py2
-            title = u'b'
-            assert_raises(TypeError, np.dtype, [(title, int)])
-            assert_raises(TypeError, np.dtype, [(('a', title), int)])
+            # Unicode field names are converted to ascii on Python 2:
+            encodable_title = u'b'
+            assert_equal(np.dtype([(encodable_title, int)]).names[0], b'b')
+            assert_equal(np.dtype([(('a', encodable_title), int)]).names[0], b'b')
+
+            # But raises UnicodeEncodeError if it can't be encoded:
+            nonencodable_title = u'\uc3bc'
+            assert_raises(UnicodeEncodeError, np.dtype, [(nonencodable_title, int)])
+            assert_raises(UnicodeEncodeError, np.dtype, [(('a', nonencodable_title), int)])
```

Comment:
Why does this fall on py3 too?

Reply:
I'm not completely sure what you're asking, but this part of `test_multiarray.py` is within an if statement such that it only is defined on Python 2, if that answers your question.

Reply:
Got it. I would have expected a `@skipIf`  decorator for that, but that's cleanup for another time.